### ISSUE TITLE
Set TileJSON layer metadata for function sources in config

### ISSUE THREAD

**Issue by nick-romano** [urbica/martin#822]
For full parity with table sources, it would be helpful to have metadata available for function sources within the tilejson response.

Some relevant issues:
- https://github.com/maplibre/martin/issues/749
- https://github.com/maplibre/martin/issues/277

**Comment by nick-romano** [comment]
seems like this could be difficult if the function accepts query params...

**Comment by nyurik** [comment]
We definitely need to be able to set various per-source settings in the config, especially settings that are not auto-discoverable from the source itself.  Let's sketch out some config structures first - my proposal is to allow direct tilejson modification:

```yaml
postgres:
  functions:
    my_func:
      tilejson:
        description: my magical tile source
        # Here goes any fields that should be set in the tilejson,
        # overriding anything that's already being set there automatically.
        # Note that setting field to `null`, e.g. `description: ~` should remove it from output.
```

Note that the above could be used for all other sources, not just Postgres functions (i.e. tables, mbtiles, pmtiles). Moreover, it could be allowed in the `auto_publish` section too.  In auto-publish, it could use the same placeholders as in the `source_id_format`, e.g. `{schema}, {table}, {column}`

**Comment by nyurik** [comment]
We could also introduce an optional tilejson function that gets called on `/source`. It would support two function types:
* `FUNCTION get_meta(source_id TEXT, host TEXT) RETURNS JSON`
* `FUNCTION get_meta(source_id TEXT, host TEXT, query_params JSON) RETURNS JSON`

Passing `source_id` to the function allows the same function to be used for multiple sources.  The `host` param is taken from the `Host` header of the rquest - allows tilejson to be re-configured for another domain request as needed (this is done automatically too)

Moreover, it could be used with a table source too.  The returned JSON has the same content as the `tilejson` config key above.  In the config, it could be specified as `tilejson_function: get_meta`.  Having both `tilejson` and `tilejson_function` would merge `autogenerated + tilejson + function call`.

**Comment by sharkAndshark** [comment]
Could we move layers metadata of function from martin config  to comment on function in PostgreSQL,
Like Let martin discover layer infos from  comment of function.

CREATE DEMO FUNCTION

```SQL
CREATE OR REPLACE FUNCTION osm_trees(z INTEGER, x INTEGER,y INTEGER)
RETURNS INTEGER AS $$
	select z+x+y;
$$ 
LANGUAGE SQL;
```

COMMENT TILE LAYERS INFO ON FUNCTION
```SQL
comment on function osm_trees(INTEGER,INTEGER,INTEGER) IS $tilejson$
{
    "tilejson": "3.0.0",
    "name": "OpenStreetMap",
    "description": "A free editable map of the whole world.",
    "version": "1.0.0",
    "attribution": "(c) OpenStreetMap contributors, CC-BY-SA",
    "scheme": "xyz",
    "minzoom": 0,
    "maxzoom": 18,
    "bounds": [ -180, -85, 180, 85 ],
    "fillzoom": 6,
    "something_custom": "this is my unique field",
    "vector_layers": [
        {
            "id": "telephone",
            "fields": {
                "phone_number": "the phone number",
                "payment": "how to pay"
            }
        },
        {
            "id": "bicycle_parking",
            "fields": {
                "type": "the type of bike parking",
                "year_installed": "the year the bike parking was installed"
            }
        },
        {
            "id": "showers",
            "fields": {
                "water_temperature": "the maximum water temperature",
                "wear_sandles": "whether you should wear sandles or not",
                "wheelchair": "is the shower wheelchair friendly?"
            }
        }
    ]
}
$tilejson$
```

RETRIEVE THE COMMENT
```SQL
select "pg_description"."description" from "pg_proc" 
left join "pg_description" on "pg_description"."objoid" = "pg_proc".oid
where "pg_proc".proname = 'osm_trees'

```

**Comment by nyurik** [comment]
That's an interesting idea @sharkAndshark , I wasn't aware of metadata storage. While totally doable, I am not certain of the value to the users:
* it's not guaranteed to be valid json, whereas a metadata function ensures the result is valid
* it does not support query parameters, e.g. `http://localhost/source?param1=10` will not be able to dynamically generate a different tilejson response (if needed)

In short, it has nearly the same complexity as having a dedicated metadata-retrieving function.

**Comment by nyurik** [comment]
On second thought, a comment-approach benefit is that PostgreSQL provides automatic association between the function and the comment, thus Martin will be able to discover it without any additional configuration changes.  Moreover, this might be the fastest path to get done...

@sharkAndshark sure, agree, let's do it too :)  Any takers?

P.S. in documentation, we should show how to generate a JSON content with validation, e.g. use `execute`:

```
EXECUTE format("comment on function osm_trees(INTEGER,INTEGER,INTEGER) IS $tilejson$%s$tilejson$",
'{ "description": "my description" }'::json)
```

**Comment by sharkAndshark** [comment]
Yeah sure, I will give it a try.

**Comment by CommanderStorm** [comment]
Think this was resolved via
- #829
